### PR TITLE
Phil debug

### PR DESCRIPTION
**Pull request by tomyhoi**
Runtime:
    Change to utilize the task callback layer.
    Move all code to write files to runtime/src/chpl-visual-debug.c, other files in runtime just call 
            functions in runtime/src/chpl-visual-debug.c.
    Remove visual-debug code from tasking code.

module/standard/VisualDebug.chpl:
     Change from a coforall to a hypercube broadcast procedure to start VisualDebug code on the
            locales.

Chplvis:
     Rework DataModel and ViewField so DataModel does all the summary processing of events and
          view field.
     Add task Begin and End events to Event.
     Bug fixes.

### CODE REVIEW

File: modules/standard/VisualDebug.chpl [hunk 1]
```diff
@@ -53,25 +53,39 @@ module VisualDebug
 pragma "no doc"
   enum vis_op {v_start, v_stop, v_tag, v_pause};
 
-  private proc VDebugTree (what: vis_op, name: string, time: real, id: int = 0) {
-      var child = id * 2 + 1;
+pragma "no doc"
```
Comment (ronawho) [inline modules/standard/VisualDebug.chpl]: Is there any reason for the `pragma "no doc"` instead of just making this a private iterator?

**Comment by ronawho** [comment]
Overall, looks good to me!

I didn't review the data model or the gui parts very carefully. I'm happy to review more thoroughly if you want me to, but I figured you mostly wanted a review of the runtime changes.

File: modules/standard/VisualDebug.chpl (hunk 1 above)
Comment (tomyhoi) [inline modules/standard/VisualDebug.chpl]: No reason, I just missed it.

File: modules/standard/VisualDebug.chpl [hunk 2]
```diff
@@ -53,25 +53,39 @@ module VisualDebug
...
+iter hc_id2com ( id: int, off: int ) {
```
Comment (ronawho) [inline modules/standard/VisualDebug.chpl]: I think `hc_id2com` and `VDebugTree` could use a minor comment to describe what they do / what their purpose is. It took me a while to figure it out

File: modules/standard/VisualDebug.chpl [hunk 3]
```diff
@@ -53,25 +53,38 @@ module VisualDebug
 pragma "no doc"
   enum vis_op {v_start, v_stop, v_tag, v_pause};
 
-  private proc VDebugTree (what: vis_op, name: string, time: real, id: int = 0) {
-      var child = id * 2 + 1;
+private iter hc_id2com ( id: int, off: int ) {
```
Comment (ronawho) [inline modules/standard/VisualDebug.chpl]: I think hc_id2com and VDebugTree could use a minor comment to describe what they do / what their purpose is. It took me a while to figure it out

File: modules/standard/VisualDebug.chpl [hunk 4]
```diff
@@ -53,25 +53,38 @@ module VisualDebug
 pragma "no doc"
```
Comment (ronawho) [inline modules/standard/VisualDebug.chpl]: Can vis_op be private too?

File: runtime/src/chpl-visual-debug.c [hunk 5]
```diff
@@ -342,5 +355,62 @@ void chpl_vdebug_log_task_queue(chpl_fn_int_t     fid,
...
+    (void) chpl_task_uninstall_callback(chpl_task_cb_event_kind_begin, cb_task_create_begin);
+    return 1;
+  }
+  return 0;
+}
+
+int uninstall_callbacks(void) {
+  int rv = 0;
+  rv = chpl_task_uninstall_callback(chpl_task_cb_event_kind_create, cb_task_create_begin);
+  rv = chpl_task_uninstall_callback(chpl_task_cb_event_kind_begin, cb_task_create_begin);
```
Comment (ronawho) [inline runtime/src/chpl-visual-debug.c]: should this one use == too?

File: modules/standard/VisualDebug.chpl (hunk 4 above)
Comment (tomyhoi) [inline modules/standard/VisualDebug.chpl]: It appears that enums can't be private.  I got a compile error.

**Pull request merged**